app/backend/src/tests/test_communities.py

```diff
@@ -869,6 +881,51 @@ def test_LeaveCommunity_regression(db):
         assert not api.GetCommunity(communities_pb2.GetCommunityReq(community_id=c2_id)).member
 
 
+def test_enforce_community_memberships_for_user(testing_communities):
+    """
+    Make sure the user is added to the right communities on signup
+    """
+    with auth_api_session() as (auth_api, metadata_interceptor):
+        res = auth_api.SignupFlow(
+            auth_pb2.SignupFlowReq(
+                basic=auth_pb2.SignupBasic(name="testing", email="email@couchers.org.invalid"),
+                account=auth_pb2.SignupAccount(
+                    username="frodo",
+                    birthdate="1970-01-01",
+                    gender="Bot",
+                    hosting_status=api_pb2.HOSTING_STATUS_CAN_HOST,
+                    city="Country 1, Region 1, City 2",
+                    # == create_coordinate(8)
+                    lat=8,
+                    lng=1,
+                    radius=500,
+                    accept_tos=True,
+                ),
+                feedback=auth_pb2.ContributorForm(),
+            )
+        )
+    with session_scope() as session:
+        email_token = (
+            session.execute(select(SignupFlow).where(SignupFlow.flow_token == res.flow_token)).scalar_one().email_token
+        )
+    with auth_api_session() as (auth_api, metadata_interceptor):
+        res = auth_api.SignupFlow(auth_pb2.SignupFlowReq(email_token=email_token))
+    user_id = res.auth_res.user_id
+
+    # now check the user is in the right communities
+    with session_scope() as session:
+        w_id = get_community_id(session, "Global")
+        c1_id = get_community_id(session, "Country 1")
+        c1r1_id = get_community_id(session, "Country 1, Region 1")
+        c1r1c2_id = get_community_id(session, "Country 1, Region 1, City 2")
+
+    token = get_session_cookie_token(metadata_interceptor)
+
+    with communities_session(token) as api:
+        res = api.ListUserCommunities(communities_pb2.ListUserCommunitiesReq())
```

Comment:
Why don't you need like empty here?

Reply:
`ListUserCommunitiesReq` optionally takes pagination stuff

Reply:
Isn't it still empty? Just trying to understand, because I've had api calls fail for lack of a request before.

Reply:
I think it needs something inside `ListUserCommunities(...)`, the grpc python client might be picky towards what you put in there, but on the wire; if this is empty, it'd be equivalent to `api.ListUserCommunities(empy_pb2.Empty())`.